This patch enables aufs to use the splice functions exported by the
unionfs kernel patch.

diff -Nru aufs/fs/aufs/vfsub.h aufs.new/fs/aufs/vfsub.h
--- aufs/fs/aufs/vfsub.h	2008-02-03 22:22:46.000000000 -0600
+++ aufs/fs/aufs/vfsub.h	2008-03-26 20:51:04.000000000 -0500
@@ -604,7 +604,7 @@
 		  AuDLNPair(in->f_dentry), *ppos, (unsigned long)len, flags);
 
 	lockdep_off();
-	err = do_splice_to(in, ppos, pipe, len, flags);
+	err = vfs_splice_to(in, ppos, pipe, len, flags);
 	lockdep_on();
 	if (err >= 0)
 		au_update_fuse_h_inode(in->f_vfsmnt, in->f_dentry); /*ignore*/
@@ -621,7 +621,7 @@
 		  AuDLNPair(out->f_dentry), *ppos, (unsigned long)len, flags);
 
 	lockdep_off();
-	err = do_splice_from(pipe, out, ppos, len, flags);
+	err = vfs_splice_from(pipe, out, ppos, len, flags);
 	lockdep_on();
 	if (err >= 0)
 		au_update_fuse_h_inode(out->f_vfsmnt, out->f_dentry); /*ignore*/
